### ISSUE TITLE
Issues Recreating Paper Performance

### ISSUE THREAD

**Issue by AbhinavPeri** [AbhinavPeri/DeepCDR-Reimplementation#1]
**Problem:**
The model is not reaching an MSE loss of 1.16 as described in the paper. The closest it has gotten is 2. Some Optuna tuned models reached 1.16 for the validation data, but the test loss was around 6 which implies some kind of issue with the data splits.

**What am I trying:**

- When doing the data splits, I took the test set from the end of the dataset which means it might not have the same composition as the randomly created training and validation splits. As a result, I'm making the test set also randomized. This is also what the code from the paper does, randomly generating validation and training splits, and using the validation split as their testing set.
- Looked at the model structure, and saw that they were using Global Max Pooling as well several Linear Layers after the encoders, so I added those in
- Currently training a model with the best model parameters observed from Optuna hyperparameter tuning with these changes for 100 epochs.

**Comment by AbhinavPeri** [comment]
Using Global Max Pooling and Linear layers after the encoders fixed the issues with model loss. There were a couple bugs with the split generation which were also fixed. Currently doing repeat trials to get a mean and standard deviation.
